### ISSUE TITLE
how to reproduce fine-tune result？

### ISSUE THREAD

**Issue by LinB203** [THUDM/ImageReward#34]
**Can I use following code to reproduce your fine-tuning?**

from ImageReward import ReFL
args = ReFL.parse_args()
trainer = ReFL.Trainer("CompVis/stable-diffusion-v1-4", "data/refl_data.json", args=args)
trainer.train(args=args)

**Comment by xujz18** [comment]
You can refer #24

**Comment by LinB203** [comment]
> You can refer #24

Thanks for your reply. Do the reward loss and pretrain loss are trained sequentially?

**Comment by xujz18** [comment]
Just refer to the Algorithm:
![图片2](https://github.com/THUDM/ImageReward/assets/52482160/a3e138c9-8347-4f31-80d1-ce3cd5fd9d6b)

**Comment by LinB203** [comment]
> Just refer to the Algorithm: ![图片2](https://user-images.githubusercontent.com/52482160/252149464-a3e138c9-8347-4f31-80d1-ce3cd5fd9d6b.png)

Clearly. And I wonder that do you plan to release training code of reward model？

**Comment by xujz18** [comment]
You can refer https://github.com/THUDM/ImageReward#training-code-for-imagereward

**Comment by LinB203** [comment]
> You can refer https://github.com/THUDM/ImageReward#training-code-for-imagereward

Sorry, thanks for your patient reply.

**Comment by hkunzhe** [comment]
> Just refer to the Algorithm: ![图片2](https://user-images.githubusercontent.com/52482160/252149464-a3e138c9-8347-4f31-80d1-ce3cd5fd9d6b.png)

I want to add pre-training data myself. If we don't use the gradient accumulation, the pseudo code would be like this:

```python
# Given optimizer and lr_scheduler with unet.
# Compute Pre-training Loss `train_loss` with unet and update unet.
train_loss.backward()
optimizer.step()
lr_scheduler.step()
optimizer.zero_grad()

# Compute ReFL Loss `refl_loss` with unet and update unet.
refl_loss.backward()
optimizer.step()
lr_scheduler.step()
optimizer.zero_grad()
```

However, I'm confused about how to add `accelerator.accumulate(unet)` for gradient accumulation after reading [this post](https://discuss.pytorch.org/t/multiple-gradient-updates-with-two-separate-losses-and-two-classifiers-sharing-the-same-encoder/170207/5). Can you give me some pseudo codes or hints? Thank you very much! @xujz18.
